tests/unittests/sync_voodoo_palette_tests.py

```
@@ -0,0 +1,247 @@
+            with patch.object(fosvp, "TOKENS_CANDIDATES", []):
+                with self.assertRaises(SystemExit):
+                    fosvp._find_tokens()
+
+    def test_skip_if_missing_returns_none(self):
+        with patch.dict(os.environ, {}, clear=True):
+            with patch.object(fosvp, "TOKENS_CANDIDATES", []):
+                self.assertIsNone(fosvp._find_tokens(skip_if_missing=True))
+
+
+class MainTests(unittest.TestCase):
+    def _run(self, argv, constants_body="DEFAULT_APP_COLOR_POOL = []"):
```

Ruff (0.16.1)

[warning] Missing return type annotation for private function `_run`

(ANN202)